src/common/conditional_compilation/include/openvino/cc/pass/itt.hpp

```diff
@@ -35,9 +42,45 @@ OV_CC_DOMAINS(ov_pass);
 
 #    define RUN_ON_MODEL_SCOPE(region) MATCHER_SCOPE_(ov_pass, OV_PP_CAT(region, _run_on_model))
 
+#    define ADD_MATCHER_NO_OBJ_1(region, ...) add_matcher<region>(__VA_ARGS__);
+#    define ADD_MATCHER_NO_OBJ_0(region, ...)
+#    define ADD_MATCHER_FOR_THIS(nspace, region, ...)                                          \
+        OV_PP_CAT(ADD_MATCHER_NO_OBJ_, OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(ov_pass, _, region))) \
+        (nspace::region, __VA_ARGS__)
+#    define ADD_MATCHER_FOR_THIS_WITHOUT_NSPACE(region, ...)                                   \
+        OV_PP_CAT(ADD_MATCHER_NO_OBJ_, OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(ov_pass, _, region))) \
+        (region, __VA_ARGS__)
+#    define ADD_MATCHER_OBJ_1(obj, region, ...) obj->add_matcher<region>(__VA_ARGS__);
+#    define ADD_MATCHER_OBJ_0(obj, region, ...)
+#    define ADD_MATCHER(obj, nspace, region, ...)                                           \
+        OV_PP_CAT(ADD_MATCHER_OBJ_, OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(ov_pass, _, region))) \
+        (obj, nspace::region, __VA_ARGS__)
+#    define ADD_MATCHER_WITHOUT_NSPACE(obj, region, ...)                                    \
+        OV_PP_CAT(ADD_MATCHER_OBJ_, OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(ov_pass, _, region))) \
+        (obj, region, __VA_ARGS__)
+#    define REGISTER_PASS_1(obj, region, ...) obj.register_pass<region>(__VA_ARGS__);
+#    define REGISTER_PASS_0(obj, region, ...)
+#    define REGISTER_PASS(obj, nspace, region, flag, ...)                                                  \
+        OV_PP_CAT(REGISTER_PASS_, OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(ov_pass, _, OV_PP_CAT(region, flag)))) \
+        (obj, nspace::region, __VA_ARGS__)
+#    define REGISTER_PASS_WITH_FALSE_1(obj, region, ...) obj.register_pass<region, false>(__VA_ARGS__);
+#    define REGISTER_PASS_WITH_FALSE_0(obj, region, ...)
+#    define REGISTER_DISABLED_PASS(obj, nspace, region, ...)                                         \
+        OV_PP_CAT(REGISTER_PASS_WITH_FALSE_, OV_CC_SCOPE_IS_ENABLED(OV_PP_CAT3(ov_pass, _, region))) \
+        (obj, nspace::region, __VA_ARGS__)
+#    define REGISTER_PASS_MODEL_IF(region) \
```

Comment:
Why do we need this strange macro? Looks like we can use combination of `register_pass` and `add_matcher` instead.